src/live/handleAlbumChat.js

```diff
@@ -1,11 +1,37 @@
-function handleAlbumChat(io, socket, data) {
-  const { message, userId, albumId } = data;
+const ChatMessage = require('../models/ChatMessage');
 
-  // Create a chat room name for the album chat
-  const chatRoomName = `chat:album:${albumId}`;
+async function handleAlbumChat(io, socket, data) {
+  try {
+    const { albumId } = data;
 
-  // Send the message to the room
-  socket.to(chatRoomName).emit('chat:album', { message, userId, albumId });
+    // Create a chat room name for the album chat
+    const chatRoomName = `chat:album:${albumId}`;
+
+    const msg = await ChatMessage.create({ ...data, username: data.user });
```

Comment:
`data` is coming from the browser and looks like:

```json
{
    "message": "hi",
    "albumId": "6r1lh7fHMB499vGKtIyJLy",
    "user": "64ef5e90086e1c19b888c72c",
    "isOwnMessage": true
}
```

You can see here that `isOwnMessage` will be ignored by mongo because it is not part of the `ChatMessage` schema. And `username` will also be ignored because it is not part of the schema.

The `ChatMessage` schema expects three fields to be passed in: `message` (a string), `albumId` (a string), and `user` (a Mongo ObjectID for a particular User model). 

In this case, we don't care if `isOwnMessage` is ignored (see my review comments in your frontend repo). And all the other fields that the frontend is passing in already have the correct field names that match the `ChatMessage` schema, so we can pass the `data` object directly to create a new `ChatMessage`:  
 

```suggestion
    const msg = await ChatMessage.create(data);
```